src/components/Table/components/HeaderCell/CellContent.jsx

```diff
@@ -7,16 +7,23 @@ import HeaderCellMenu from "./HeaderCellMenu";
 const CellContent = ({
   children,
   isSortable,
+  isAddEnabled,
   sortedInfo,
   onSort,
   isHidable = true,
   onColumnHide,
+  onAddColumn,
+  onColumnDelete,
   column,
   ...headerProps
 }) => {
   const isColumnHidable = isHidable && isPresent(onColumnHide);
+  const isColumnDeletable = isPresent(onColumnDelete) && column?.isDeletable;
```

Comment:
Do we need an optional chaining for `column`? 

Also, I can see that some props (eg: `isHidable`) are extracted from the column object and passed to this component, but some props(eg: `isDeletable`) are extracted from the `column` prop passed to this component. See if it can be made consistent.